Rework decK documentation

mheap opened the pull request:
### Description

**Opening as a draft to get feedback from CX. Do not merge**

Rebuild the decK documentation from scratch:

* Unversioned content
* Concrete usage examples

### Testing instructions

Preview link: 

### Checklist 

- [x] Review label added <!-- (see below) -->
- [x] [Conditional version tags](https://docs.konghq.com/contributing/conditional-rendering/#conditionally-render-content-by-version) added, if applicable.

lena-larionova commented on app/deck/file/index.md:
```diff
@@ -0,0 +1,31 @@
+title: decK File
+---
+
+decK's declarative configuration format is the canonical representation of a {{ site.base_gateway }} configuration in text form.
```
This whole doc refers to "declarative configuration", but we're essentially avoiding using this term to refer to decK files now, correct? Since "declarative configuration" should refer to non-decK declarative config for Gateway in DB-less mode.

Double-checking before I go and change every instance to "state file", in case I'm missing something here.

mheap replied:
I've always understood declarative configuration to refer to decK format, and DB-less to refer to the other config format.

DB-less format is becoming less important and we want to position deck as the one true configuration format. With that in mind, "declarative configuration" means more to a new person than "state file" imo

mheap replied:
This is an opinion btw, not a statement ^. Happy to change the terminology if it's going to be confusing

lena-larionova replied:
Colloquially, yes - it's kind of like using "Kong" to mean Kong Gateway when we all know we shouldn't, but is the normal/expected way anyway. 

If we want to continue using "declarative config" to refer to decK files (and I think we should, it's entrenched already), then we should consciously refer to DB-less config as "DB-less declarative config". There also needs to be some acknowledgement of the differences between the two - in the FAQ? On the decK landing page, even? Something that makes a nod to the fact that both configs are declarative, yes, but the file format isn't exactly the same and isn't interchangeable.

We've had customers/support occasionally complain about this in the past; it's rare, since, as you said, DB-less format is way less common - but it does happen, and it's always confusing when they run into it.

For the current language used for DB-less:
https://docs.konghq.com/gateway/latest/reference/cli/#kong-config
https://docs.konghq.com/gateway/latest/production/deployment-topologies/db-less-and-declarative-config/#main

lena-larionova replied:
I guess what I'm really saying is that DB-less config was called just "declarative config" first, and has been around longer. Doesn't mean that it should get to keep the term, but we should be aware of it and get ahead of the confusion.

mheap replied:
Agree - will update the FAQ. I think deck format as `declarative config` is the future

mheap replied:
I've updated the deck overview page. There was already an FAQ item

lena-larionova commented on app/deck/file/openapi2kong.md:
```diff
@@ -0,0 +1,94 @@
+
+`openapi2kong` allows you to configure plugins directly in your OpenAPI specification by providing an `x-kong-plugin-PLUGIN-NAME` annotation at the root level to add it to the service, at the path level to add it to all routes on that path, or on an operation to add it to a specific route.
+
+_However_, we do not recommend using this functionality. Routing and policy enforcement are generally owned by different teams, and you do not want to publish plugin configuration in an OpenAPI specification that is shared with customers.
```
Should we even document this if we don't want people using it?

mheap replied:
Sadly it's used enough that if we don't document the functionality we'll get questions about it

lena-larionova commented:
I really like this direction! Removing versioning feels right, there's no reason to keep old decK versions or docs for them.

I did a surface-level review for now, curious if there will be any external feedback, especially on removing the reference pages for the CLI flags. Those pages were mainly a way to evaluate capability though, and although people use them now, that might just be because there's nothing else.

Terminology/language wise, the main thing that jumped out at me was the use of "declarative configuration" - we should avoid using that term to prevent confusion with the kong DB-less config files, managed via the `kong config` CLI. 

Re: missing content:
* This page is missing: https://docs.konghq.com/deck/latest/3.0-upgrade/ - was it left out on purpose? Might be good to leave it for now until we exit 2.8 LTS support.
* https://docs.konghq.com/deck/latest/guides/konnect/#troubleshoot - most of this doc doesn't need to exist in this state, but the troubleshooting info should live somewhere
* https://docs.konghq.com/deck/latest/reference/deck_completion/ - this command is missing, but I don't quite understand what it does.
* some to-dos left in a couple places, but you know that :)

Would like to test some of these flows but that would take quite a while, so that's better served leaving till dev site migration.

Guaris commented:
> I really like this direction! Removing versioning feels right, there's no reason to keep old decK versions or docs for them.
I agree. 

> Re: missing content:
> 
> * This page is missing: https://docs.konghq.com/deck/latest/3.0-upgrade/ - was it left out on purpose? Might be good to leave it in for now until we exit 2.8 LTS support.
I think the FAQ is the most useful thing here to me. But I don't see any reason to remove the entire doc. 

> * https://docs.konghq.com/deck/latest/guides/konnect/#troubleshoot - most of this doc doesn't need to exist in this state, but the troubleshooting info should live somewhere
Maybe FAQs on the deck landing page?

> * https://docs.konghq.com/deck/latest/reference/deck_completion/ - this command is missing, but I don't quite understand what it does.
This command adds auto complete (so you can hit tab and get the rest of the command) to different shells.

cloudjumpercat commented on app/deck/configuration.md:
````diff
@@ -0,0 +1,50 @@
+export DECK_KONNECT_CONTROL_PLANE_NAME=demo
+export DECK_KONNECT_TOKEN=$KONNECT_TOKEN
+deck gateway sync kong.yaml
+```
````
I'd like to see an example of how these decK env var can be referenced in a decK file. Ex. `${{ env "DECK_API_KEY" }}`

mheap replied:
You wouldn't refer to these specific values in a decK file as they're used to configure the CLI.

Does http://localhost:8888/deck/gateway/sensitive-data/#using-deck-environment-variables meet your needs?

cloudjumpercat commented on app/deck/manage-gateway/diff.md:
```diff
@@ -0,0 +1,50 @@
+---
+title: diff
```
Here and for other titles in the Managing Kong Gateway section, I'd have a title that tells a new user what this is doing.
```suggestion
title: Show differences in configuration (`deck gateway diff`)
```

cloudjumpercat commented on app/_redirects:
```diff
@@ -256,13 +256,15 @@
+/deck/installation/                   /deck/installation/
+/deck/terminology/                    /deck/terminology/
+/deck/latest/ /deck/
+/deck/latest/* /deck/:splat
```
I got this 404 page from a search: `/deck/guides/environment-variables/`

mheap replied:
Search isn't expected to work until the PR is merged I believe.

mheap commented:
> This page is missing: https://docs.konghq.com/deck/latest/3.0-upgrade/ - was it left out on purpose? Might be good to leave it in for now until we exit 2.8 LTS support.

Added to reference

> https://docs.konghq.com/deck/latest/guides/konnect/#troubleshoot - most of this doc doesn't need to exist in this state, but the troubleshooting info should live somewhere

Added to the new "Konnect configuration" page


> https://docs.konghq.com/deck/latest/reference/deck_completion/ - this command is missing, but I don't quite understand what it does.

Most people don't use it, and those that need it will know how it works. I'm choosing not to document to reduce the amount of pages to look at

mheap commented:
@lena-larionova @cloudjumpercat All CI passing - time for final review please 🥳

lena-larionova commented on app/deck/gateway/configuration.md:
```diff
@@ -0,0 +1,137 @@
+---
+title: Configuration
```
```suggestion
title: Configuring Authentication
```
"Configuration" is much too general a title and tells me nothing about this page.

lena-larionova commented on app/deck/configuration.md:
```diff
@@ -0,0 +1,50 @@
+---
+title: Configuration
```
```suggestion
title: decK Configuration
```
For better search results.

lena-larionova commented on app/deck/get-started.md:
```diff
@@ -0,0 +1,127 @@
+---
+title: Get Started
```
```suggestion
title: Get Started with decK
```

lena-larionova commented on app/deck/index.md:
```diff
@@ -0,0 +1,28 @@
+
+The `deck` format is the canonical declarative configuration format in this documentation. When the words "declarative configuration" are used, it is to refer to a `deck` format file, _not_ a `dbless` format.
+
+For more information, see the [FAQ](/deck/reference/faq/#kong-already-has-built-in-declarative-configuration-do-i-still-need-deck)
```
```suggestion
For more information, see the [FAQ](/deck/reference/faq/#kong-already-has-built-in-declarative-configuration-do-i-still-need-deck).
```

lena-larionova commented on app/deck/gateway/konnect-configuration.md:
```diff
@@ -0,0 +1,134 @@
+{{site.base_gateway}} supports AU, EU, IN, ME, and US [geographic regions](/konnect/geo/).
+
+To target a specific geo, set `konnect-addr` to one of the following:
+* AU geo:`"https://au.api.konghq.com"`
+* EU geo:`"https://eu.api.konghq.com"`
+* US geo:`"https://us.api.konghq.com"`
+* IN geo:`"https://in.api.konghq.com"`
+* ME geo:`"https://me.api.konghq.com"`
+
```
```suggestion
* AU geo: `"https://au.api.konghq.com"`
* EU geo: `"https://eu.api.konghq.com"`
* US geo: `"https://us.api.konghq.com"`
* IN geo: `"https://in.api.konghq.com"`
* ME geo: `"https://me.api.konghq.com"`

```

lena-larionova commented on app/deck/gateway/ping.md:
```diff
@@ -0,0 +1,7 @@
+title: Check decK connectivity
+---
+
+The `deck file ping`command checks that `deck` can contact the Admin API. This could be either the Konnect API, or an on-prem installation.
```
```suggestion
The `deck file ping` command checks that decK can contact the Admin API. This could be either the {{site.konnect_short_name}} API, or an on-prem installation.
```

lena-larionova commented on app/_data/docs_nav_deck.yml:
```diff
@@ -0,0 +1,123 @@
+      url: /gateway/
+    - text: Konnect Configuration
+      url: /gateway/konnect-configuration/
+    - text: Configuration
```
```suggestion
    - text: Configure Authentication
```

lena-larionova commented on app/deck/gateway/diff.md:
```diff
@@ -0,0 +1,50 @@
+---
+title: Show differences in configuration (`deck gateway diff`)
```
```suggestion
title: Show differences in configuration
```
I'd only include `deck gateway diff` in the title if we were going to do that in all of these pages. Otherwise, it stands out as a bit random. Also, backticks don't work in titles.

lena-larionova commented on app/deck/apiops/continuous-integration.md:
```diff
@@ -0,0 +1,26 @@
+---
+title: Continuous Integraion
```
```suggestion
title: Continuous Integration
```

lena-larionova commented on app/deck/file/openapi2kong.md:
```diff
@@ -0,0 +1,94 @@
+
+The Kong `request-validator` plugin can validate incoming requests against a user-provided JSON schema.
+
+The plugin can be configured manually, but `openapi2kong` can use the provided OpenAPI specification to configure the `request-validator` plugin for all routes automatically. To do enable this behavior, add the `x-kong-plugin-request-validator` at the root level to add it to all routes, at the path level to add it to all routes on that path, or on an operation to add it to a specific route.
```
```suggestion
The plugin can be configured manually, but `openapi2kong` can use the provided OpenAPI specification to configure the `request-validator` plugin for all routes automatically. To enable this behavior, add the `x-kong-plugin-request-validator` at the root level to add it to all routes, at the path level to add it to all routes on that path, or on an operation to add it to a specific route.
```

lena-larionova commented on app/_data/landing_page.yml:
```diff
@@ -63,10 +63,10 @@ features:
+            url: /deck/get-started/
           - text: APIOps with decK
             icon: /assets/images/icons/icn-doc.svg
             url: /deck/latest/guides/apiops/
```
```suggestion
            url: /deck/apiops/
```

lena-larionova commented on app/deck/index.md:
```diff
@@ -0,0 +1,28 @@
+
+{{ site.base_gateway }} has historically offered two declarative configuration formats, `deck` and `dbless`. The `dbless` format is an internal format used by `kong config db_export` and {{ site.kic_product_name }}.
+
+The `deck` format is the canonical declarative configuration format in this documentation. When the words "declarative configuration" are used, it is to refer to a `deck` format file, _not_ a `dbless` format.
```
```suggestion
The `deck` format is the canonical declarative configuration format in this documentation. When the words "declarative configuration" are used, they refer to a `deck` format file, _not_ a `dbless` format.
```
